authorize_error.go

```diff
@@ -34,47 +34,49 @@ func (f *Fosite) WriteAuthorizeError(rw http.ResponseWriter, ar AuthorizeRequest
 	rw.Header().Set("Pragma", "no-cache")
 
 	rfcerr := ErrorToRFC6749Error(err)
+	if !f.SendDebugMessagesToClients {
+		rfcerr = rfcerr.Sanitize()
+	}
+
 	if !ar.IsRedirectURIValid() {
-		if !f.SendDebugMessagesToClients {
-			rfcerr.Debug = ""
-		}
+		rw.Header().Set("Content-Type", "application/json;charset=UTF-8")
 
 		js, err := json.MarshalIndent(rfcerr, "", "\t")
 		if err != nil {
-			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			if f.SendDebugMessagesToClients {
+				http.Error(rw, fmt.Sprintf(`{\n\t"error": "server_error",\n\t"error_description": "%s"\n}`, err.Error()), http.StatusInternalServerError)
```

Comment:
Same problem with potential JSON escaping as above!

```suggestion
				http.Error(rw, fmt.Sprintf(`{"error": "server_error","error_description": "%s"}`, err.Error()), http.StatusInternalServerError)
```